src/main/scala/org/scalameter/Executor.scala

```diff
@@ -1,26 +1,28 @@
 package org.scalameter
 
 
+import org.scalameter.picklers.Pickler
 import org.scalameter.utils.Tree
+import scala.language.higherKinds
 
 
 
-trait Executor {
+trait Executor[V] {
 
-  def measurer: Measurer
+  def measurer: Measurer[V]
 
   def warmer: Warmer
 
-  def run[T](setuptree: Tree[Setup[T]], reporter: Reporter, persistor: Persistor): Tree[CurveData] = {
+  def run[T](setuptree: Tree[Setup[T]], reporter: Reporter[V],
+    persistor: Persistor): Tree[CurveData[V]] = {
     for (setup <- setuptree) yield {
-      val exec = Option(setup.customExecutor).getOrElse(this)
-      val cd = exec.runSetup(setup)
+      val cd = runSetup(setup)
```

Comment:
Interesting. A quick search through the code says we're not using the `customExecutor` field any more. This means that the change in this line should not break anything. I see that you also removed it below - nice catch.